src/sage/schemes/elliptic_curves/hom_sum.py

```diff
@@ -0,0 +1,644 @@
+        """
+        return self._phis
+
+    @cached_method
+    def to_isogeny_chain(self):
+        r"""
+        Convert this formal sum of elliptic-curve morphisms into a
+        :class:`EllipticCurveHom_composite` object representing the
+        same morphism.
+
+        EXAMPLES::
+
+            sage: E = EllipticCurve(GF(101), [5,5])
+            sage: phi = E.isogenies_prime_degree(7)[0]
+            sage: (phi + phi).to_isogeny_chain()
+            Composite morphism of degree 28 = 4*1*7:
+              From: Elliptic Curve defined by y^2 = x^3 + 5*x + 5 over Finite Field of size 101
+              To:   Elliptic Curve defined by y^2 = x^3 + 12*x + 98 over Finite Field of size 101
+
+        ::
+
+            sage: p = 419
+            sage: E = EllipticCurve(GF(p^2), [1,0])
+            sage: iota = E.automorphisms()[2]   # sqrt(-1)
+            sage: pi = E.frobenius_isogeny()    # sqrt(-p)
+            sage: endo = iota + pi
+            sage: endo.degree()
+            420
+            sage: endo.to_isogeny_chain()
+            Composite morphism of degree 420 = 4*1*3*5*7:
+              From: Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 419^2
+              To:   Elliptic Curve defined by y^2 = x^3 + x over Finite Field in z2 of size 419^2
+
+        The decomposition is impossible for the constant zero map::
+
+             sage: endo = iota*pi + pi*iota
+             sage: endo.degree()
+             0
+             sage: endo.to_isogeny_chain()
+             Traceback (most recent call last):
+             ...
+             ValueError: zero morphism cannot be written as a composition of isogenies
+
+        Isomorphisms are supported as well::
+
+            sage: E = EllipticCurve(j=5); E
+            Elliptic Curve defined by y^2 + x*y = x^3 + x^2 + 180*x + 17255 over Rational Field
+            sage: m2 = E.scalar_multiplication(2)
+            sage: m3 = E.scalar_multiplication(3)
+            sage: (m2 - m3).to_isogeny_chain()
+            Composite morphism of degree 1 = 1^2:
+              From: Elliptic Curve defined by y^2 + x*y = x^3 + x^2 + 180*x + 17255 over Rational Field
+              To:   Elliptic Curve defined by y^2 + x*y = x^3 + x^2 + 180*x + 17255 over Rational Field
+            sage: (m2 - m3).rational_maps()
+            (x, -x - y)
+        """
+        deg = self.degree()
+        if deg.is_zero():
+            raise ValueError('zero morphism cannot be written as a composition of isogenies')
+
+        p = self.base_ring().characteristic()
+        insep = 0
+        scalar = 1  #TODO Can we detect scalar factors earlier to save some extensions below?
+
+        ker = []
+        for l,m in deg.factor():
```

Comment:
In this loop are you computing the inseparable degree of the sum?  It might be preferable to implement a method called inseparable_degree() using this sort of code and call it here?

Reply:
Excellent suggestion, this was indeed mentioned as a TODO elsewhere! Done.

Reply:
Thanks -- that was more work than I thought it would be (as always happens), so the extra work is appreciated.

Reply:
Will I be very unpopular to suggest that we now also need method(s) separable_degree()?  It's only a matter of time before some user complains it is not there, so now would be a good time.  You could then change the injective() method to checking that the separable_degree is 1.

Reply:
More importantly, I don't think it is correct that a sum is separable iff the summands are.  For example (Frob-1)+(1).  The reverse implication also fails, e.g. (Frob)+(1).  So I think that is_separable() for sums will have to do some work.

Reply:
Thanks, added `.separable_degree()`.

> More importantly, I don't think it is correct that a sum is separable iff the summands are. For example (Frob-1)+(1). The reverse implication also fails, e.g. (Frob)+(1). So I think that is_separable() for sums will have to do some work.

Certainly, but I don't think the current branch does it wrong, does it? The `.is_separable()` method inherited from `EllipticCurveHom` simply checks that the inseparable degree equals one, so having a correct implementation of the `.inseparable_degree()` method suffices, unless I'm missing something.

Reply:
Sorry, you are moving faster than I can keep up! I'm sure I did once see an implementation which just checked if the summands were seperable.  It's fine now.